### DISCUSSION TITLE
Y-Sorting?

### DISCUSSION THREAD

**Discussion by doonv** [StarArawn/bevy_ecs_tilemap#491] [Q&A]
How do I do Y-Sorting for tiles? The way I would do this with sprites is a simple system like this:
```rs
#[derive(Component)]
pub struct YSort;

pub fn y_sort(mut query: Query<&mut Transform, With<YSort>>) {
    for mut transform in query.iter_mut() {
        transform.translation.z =
            -(1.0 / (1.0 + (2.0f32.powf(-0.01 * transform.translation.y))));
    }
}
```

But just adding the `YSort` component from the above script into the tiles doesn't seem to do anything, it appears the tiles inherit the transform of their map.

Is there an example for doing Y-Sorting with tiles?

**Comment by PraxTube** [comment]
I am also looking into this right now. Seems like the [TilemapRenderSettings](https://docs.rs/bevy_ecs_tilemap/latest/bevy_ecs_tilemap/map/struct.TilemapRenderSettings.html#structfield.y_sort) is what we need here, but I am also not quite sure how do make that works nicely with the rest of the game (e.g. player sprites and so on). The y-sort was introduced in #408.

**Comment by feelingsonice** [comment]
Following, interested.
